### DISCUSSION TITLE
Encountered error while creating the IPA: error: exportArchive "Runner.app" requires a provisioning profile with the Push Notifications feature.

### DISCUSSION THREAD

**Discussion by Luqmanasif6** [codemagic-ci-cd/codemagic-docs#2806] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Building App Store IPA...                                     
Encountered error while creating the IPA:
error: exportArchive "Runner.app" requires a provisioning profile with the Push Notifications feature.
error: exportArchive "ImageNotification.appex" requires a provisioning profile with the Push Notifications feature.

### Expected results

Building App Store IPA...                                        
✓ Built IPA to build/ios/ipa
To upload to the App Store either:
    1. Drag and drop the "build/ios/ipa/*.ipa" bundle into the Apple Transporter macOS app https://apps.apple.com/us/app/transporter/id1450874784
    2. Run "xcrun altool --upload-app --type ios -f build/ios/ipa/*.ipa --apiKey your_api_key --apiIssuer your_issuer_id".
       See "man altool" for details about how to authenticate with the App Store Connect API key.

### Actual results

I am implementing white labeling in my Flutter app using `codemagic.yaml`. After successfully completing all steps in the Codemagic script, I encountered the following error when running the "Flutter build ipa and automatic version" command:

error: exportArchive "Runner.app" requires a provisioning profile with the Push Notifications feature.
error: exportArchive "ImageNotification.appex" requires a provisioning profile with the Push Notifications feature.
![Screenshot 2024-10-30 at 6 32 34 PM](https://github.com/user-attachments/assets/a208738a-7d07-4d04-8dd7-f826616cb4c2)

I’ve followed all the steps discussed on GitHub, but I’m still facing this issue. Could someone please assist me in identifying the cause and resolving this error? Your help would be greatly appreciated. Thank you!

### Build id (optional)

6722083ee5c1530343172de2

**Comment by dam-ease** [comment]
Hello @Luqmanasif6,
We currently have an ongoing thread on discord, please let's continue there.

**Reply by Luqmanasif6** [reply]
Sure! I'm Looking forward to your response there!

**Comment by jcho0216** [comment]
Updates?

**Comment by marcgc21** [comment]
I ended on the same problem. I tried to find on Discord server but without success. Someone can update with a solution?

**Reply by dam-ease** [reply]
Hey @marcgc21 @jcho0216,
Missed your messages; this was resolved by confirming the following:

- Capabilities like Push Notifications are connected to the app identifier in the provisioning profile. Please review the Apple Developer portal under the [Identifiers section](https://developer.apple.com/account/resources/identifiers/list) and confirm that the Push Notifications capability is enabled for your bundle ID.
- Double-check that the profile that holds that identifier doesn’t have an invalid label under the expiration column
![image_3](https://github.com/user-attachments/assets/d47e19e3-09cc-4e58-8372-428dac0a2c98)
![Screenshot_2024-10-21_at_09 11 21](https://github.com/user-attachments/assets/6f3ded67-0897-4f3e-819c-10f47d8854ad)

Also, sharing your buildID will also help here. You can find it under your build details page's 'ID' section.
